src/serve.rs

```diff
@@ -0,0 +1,244 @@
+    #[error("serde")]
+    Serde(#[from] serde_json::Error),
+    #[error("hex")]
+    FromHex(#[from] FromHexError),
+    #[error("contractnotfound")]
+    FunctionNotFoundInContractSpec,
+    #[error("unknownmethod")]
+    UnknownMethod,
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+#[serde(rename_all = "snake_case")]
+#[serde(untagged)]
+enum Requests {
+    Call {
+        id: String,
+        func: String,
+        args: Option<Vec<Value>>,
+        args_xdr: Option<Vec<String>>,
+    },
+}
+
+impl Cmd {
+    pub async fn run(&self) -> Result<(), Error> {
+        // let context = HostContext::default();
+        // eprintln!("Hello World!");
+        // process::exit(0);
+
+        let ledger_file = Arc::new(self.ledger_file.clone());
+        let with_ledger_file = warp::any().map(move || ledger_file.clone());
+
+        let call = warp::post()
+            .and(warp::path("rpc"))
+            .and(warp::body::json())
+            .and(with_ledger_file)
+            .map(
+                |request: jsonrpc::Request<Requests>, ledger_file: Arc<PathBuf>| {
+                    if request.jsonrpc != "2.0" {
+                        return json!({
+                            "jsonrpc": "2.0",
+                            "id": &request.id,
+                            "error": {
+                                "code":-32600,
+                                "message": "Invalid jsonrpc value in request",
+                            },
+                        })
+                        .to_string();
+                    }
+                    let result = match (request.method.as_str(), request.params) {
+                        (
+                            "call",
+                            Some(Requests::Call {
+                                id,
+                                func,
+                                args,
+                                args_xdr,
+                            }),
+                        ) => invoke(
+                            &id,
+                            &func,
+                            &args.unwrap_or_default(),
+                            &args_xdr.unwrap_or_default(),
+                            &ledger_file,
+                        ),
+                        _ => Err(Error::UnknownMethod),
+                    };
+                    let r = reply(&request.id, result);
+                    serde_json::to_string(&r).unwrap_or_else(|_| {
+                        json!({
+                            "jsonrpc": "2.0",
+                            "id": &request.id,
+                            "error": {
+                            "code":-32603,
+                            "message": "Internal server error",
+                            },
+                        })
+                        .to_string()
+                    })
+                },
+            );
+
+        let addr: SocketAddr = ([127, 0, 0, 1], self.port.unwrap_or(8080)).into();
```

Comment:
💡 Can you make the port configurable? It's likely to be a pain point as 8080 can be often already in use.